File: src/main/java/com/tobe/healthy/common/LessonTimeFormatter.java
package com.tobe.healthy.common;

import java.time.LocalDate;
import java.time.LocalTime;
import java.time.format.DateTimeFormatter;
import java.util.Locale;

public class LessonTimeFormatter {

	private LessonTimeFormatter() {
	}

	public static String formatLessonTime(LocalTime lessonStartTime, LocalTime lessonEndTime) {
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm");
		String startTime = lessonStartTime != null ? lessonStartTime.format(formatter) : null;
		String endTime = lessonEndTime != null ? lessonEndTime.format(formatter) : null;
		return startTime + " - " + endTime;
	}

	public static String formatLessonTimeWithAMPM(LocalTime lessonStartTime, LocalTime lessonEndTime) {
		DateTimeFormatter startTimeFormatter = DateTimeFormatter.ofPattern("a hh:mm");
		DateTimeFormatter endTimeFormatter = DateTimeFormatter.ofPattern("hh:mm");
		String startTime = lessonStartTime.format(startTimeFormatter);
		String endTime = lessonEndTime.format(endTimeFormatter);
		return startTime + " - " + endTime;
	}

	public static String formatLessonTime(LocalTime localTime) {
		if (localTime == null) {
			return null;
		}
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm");
		return localTime.format(formatter);
	}

	public static String formatLessonDt(LocalDate lessonDt) {
		if (lessonDt == null) {
			return null;
		}
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MM월 dd일 E요일", Locale.KOREAN);
		return lessonDt.format(formatter);
	}

	public static DateTimeFormatter lessonStartDateTimeFormatter() {
		return DateTimeFormatter.ofPattern("M월 d일(E) h시");
	}
}

File: src/main/kotlin/com/tobe/healthy/common/LessonTimeFormatter.kt
package com.tobe.healthy.common

import java.time.LocalDate
import java.time.LocalTime
import java.time.format.DateTimeFormatter
import java.util.*

object LessonTimeFormatter {
    fun formatLessonTime(lessonStartTime: LocalTime?, lessonEndTime: LocalTime?): String {
        val formatter = DateTimeFormatter.ofPattern("HH:mm")
        val startTime = lessonStartTime?.format(formatter)
        val endTime = lessonEndTime?.format(formatter)
        return "${startTime} - ${endTime}"
    }

    fun formatLessonTimeWithAMPM(lessonStartTime: LocalTime, lessonEndTime: LocalTime): String {
        val startTimeFormatter = DateTimeFormatter.ofPattern("a hh:mm")
        val endTimeFormatter = DateTimeFormatter.ofPattern("hh:mm")
        val startTime = lessonStartTime.format(startTimeFormatter)
        val endTime = lessonEndTime.format(endTimeFormatter)
        return "${startTime} - ${endTime}"
    }

    fun formatLessonTime(localTime: LocalTime?): String? {
        val formatter = DateTimeFormatter.ofPattern("HH:mm")
        return localTime?.format(formatter)
    }

    fun formatLessonDt(lessonDt: LocalDate?): String? {
        val formatter = DateTimeFormatter.ofPattern("MM월 dd일 E요일", Locale.KOREAN)
        return lessonDt?.format(formatter)
    }

    @JvmStatic
    fun lessonStartDateTimeFormatter(): DateTimeFormatter {
        return DateTimeFormatter.ofPattern("M월 d일(E) h시")
    }
}